### ISSUE TITLE
unittest_lint.py fails with IsADirectoryError

### ISSUE THREAD

**Issue by paper42** [PyCQA/pylint#4780]
### Bug description

https://github.com/PyCQA/pylint/blob/main/tests/lint/unittest_lint.py#L638
```python
try:
    pylintd = join(tempfile.gettempdir(), ".pylint.d")
    os.environ["PYLINTHOME"] = pylintd
    try:
        reload(config)
        assert config.PYLINT_HOME == pylintd
    finally:
        try:
>           os.remove(pylintd)
        except FileNotFoundError:
            pass
finally:
    del os.environ["PYLINTHOME"]
```

This test fails with IsADirectoryError, because pylintd is a directory. It can be fixed by replacing os.remove with rmtree.

### Command used

```shell
pytest -k test_pylint_home
```

### Pylint output

```python3
_______________________________ test_pylint_home _______________________________

    @pytest.mark.usefixtures("pop_pylintrc")
    def test_pylint_home():
        uhome = os.path.expanduser("~")
        if uhome == "~":
            expected = ".pylint.d"
        else:
            expected = os.path.join(uhome, ".pylint.d")
        assert config.PYLINT_HOME == expected
    
        try:
            pylintd = join(tempfile.gettempdir(), ".pylint.d")
            os.environ["PYLINTHOME"] = pylintd
            try:
                reload(config)
                assert config.PYLINT_HOME == pylintd
            finally:
                try:
>                   os.remove(pylintd)
E                   IsADirectoryError: [Errno 21] Is a directory: '/tmp/.pylint.d'

tests/lint/unittest_lint.py:645: IsADirectoryError
```

### Expected behavior

tests pass

### Pylint version

```
pylint 2.9.6
astroid 2.6.5
Python 3.9.6 (default, Jul  6 2021, 18:29:50) 
[GCC 10.2.1 20201203]
```

### OS / Environment

Void Linux

**Comment by Pierre-Sassoulas** [comment]
Thank you for the report @paper42 , would you like to contribute the fix as you already analyzed the problem ? :)

**Comment by paper42** [comment]
sure, #4781
